students/BrianNisonger/session9/test_mailroom.py

```diff
@@ -0,0 +1,181 @@
+    d_list = Donors([d1, d2, d3])
+    assert d_list.donor_list[1].donation == [500, 600, 700, 800]
+
+
+def test_list_donors():
+    d1 = Donor("Fred Flinstone")
+    d2 = Donor("James Dean")
+    d3 = Donor("Jack the Ripper")
+    d_list = Donors([d1, d2, d3])
+    list_of_donors = d_list.list_donors()
+    assert list_of_donors == [
+        "Fred Flinstone", "James Dean", "Jack the Ripper"
+    ]
+
+
+def test_sort_donors():
+    d1 = Donor("Fred Flinstone", [100, 200, 300, 400])
+    d2 = Donor("James Dean", [500, 600, 700, 800])
+    d3 = Donor("Jack the Ripper", [1, 2, 3, 4])
+    d4 = Donor("Mickey Mouse")
+    d_list = Donors([d1, d2, d3])
+    d_list.sort_donors()
+    list_of_donors = d_list.list_donors()
+    assert list_of_donors == [
+        "James Dean", "Fred Flinstone", "Jack the Ripper"
+    ]
+    d_list.append(d4)
+    d_list.sort_donors()
+    list_of_donors = d_list.list_donors()
+    assert list_of_donors == [
+        "James Dean", "Fred Flinstone", "Jack the Ripper", "Mickey Mouse"
+    ]
+
+
+def test_make_report():
+    d1 = Donor("Fred Flinstone", [100, 200, 300, 400])
+    d2 = Donor("James Dean", [500, 600, 700, 800])
+    d3 = Donor("Jack the Ripper", [1, 2, 3, 4])
+    d_list = Donors([d1, d2, d3])
+    report = d_list.make_report()
+    assert d_list.make_headers_string(
+    ) == 'Donor Names          Total Given     Num Gifts       Average Gifts  '
+    assert report.startswith('Donor')
+    assert 'Fred Flinstone       $1000.00         4               $250.00' in report
+    assert 'James Dean           $2600.00         4               $650.00' in report
+    assert 'Jack the Ripper      $10.00           4               $2.50' in report
+    assert report.endswith('\n')
+
+
+def test_make_thank_strings():
+    d1 = Donor("Fred Flinstone", [100, 200, 300, 400])
+    d2 = Donor("James Dean", [500, 600, 700, 800])
+    d3 = Donor("Jack the Ripper", [1, 2, 3, 4])
+    d_list = Donors([d1, d2, d3])
+    thank_notes = d_list.make_thank_notes()
+    assert thank_notes == [
+        'Dear Fred Flinstone, Thank you for your donation of $1000.00. These funds help save the migratory butterflies of New South East.Thank you',
+        'Dear James Dean, Thank you for your donation of $2600.00. These funds help save the migratory butterflies of New South East.Thank you',
+        'Dear Jack the Ripper, Thank you for your donation of $10.00. These funds help save the migratory butterflies of New South East.Thank you'
+    ]
+
+
+def test_find_donor():
+    d1 = Donor("Fred Flinstone", [100, 200, 300, 400])
+    d2 = Donor("James Dean", [500, 600, 700, 800])
+    d3 = Donor("Jack the Ripper", [1, 2, 3, 4])
+    d4 = Donor("Mickey Mouse", [100, 200, 300, 5, 15, 7])
+    d_list = Donors([d1, d2, d3, d4])
+    donor_class = d_list.find_donor("Mickey Mouse")
+    assert donor_class.donation == [100, 200, 300, 5, 15, 7]
+    donor_class = d_list.find_donor("")
+    assert donor_class == []
+
+
+def test_make_filenames():
```

Comment:
you are already testing the make_filename in the Donor class -- you don't need to test that here, too. -- at least not the names themselves.